testing/test_end_to_end.py

```diff
@@ -0,0 +1,42 @@
+import unittest
+import hashlib
+import glob
+
+
+class testEndToEnd(unittest.TestCase):
+
+    # Method that returns the md5 hash of a given file
+    def generate_md5(self, filename):
```

Comment:
@KyleBehiels Every function needs docstrings with param and return definitions. 

https://www.jetbrains.com/help/pycharm/creating-documentation-comments.html